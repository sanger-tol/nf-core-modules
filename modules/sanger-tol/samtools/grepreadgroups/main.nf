process SAMTOOLS_GREPREADGROUPS {
    tag "${meta.id}"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/samtools:1.22.1--h96c455f_0':
        'biocontainers/samtools:1.22.1--h96c455f_0' }"

    input:
    tuple val(meta), path(bam)

    output:
    tuple val(meta), stdout, emit: read_group
    path "versions.yml"    , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args   = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    samtools view -H ${bam} |\\
        ## Do some pretty printing with awk to avoid starting or trailing newlines
        awk '/@RG/ { printf "%s%s", (n++ ? "\\n" : ""), \$0 }'

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//' )
    END_VERSIONS
    """

    stub:
    def args   = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    echo -n "@RG	ID:49240_6#4	DT:2024-07-25T16:52:42Z	PU:20240725_LH00275_0063_A22CHVNLT4_6#4	LB:72611119	PG:SCS	SM:SAMEA111361848	CN:SC	PL:ILLUMINA	DS:ERP129860: Sequencing and assembly of symbiont/host genomes in marine and freshwater species, where at least one partner is a microbe, for the Aquatic Symbiosis Genomics project. Upon completion, genomes will be publicly available on an online data coordination center run by the European Bioinformatics Institute. All the data the project generates will be released openly and fairly, as the effort works towards a global community of scientists and ecologists aiming to understand and conserve ocean and freshwater biodiversity. This data is part of a pre-publication release. For information on the proper use of pre-publication data shared by the Wellcome Trust Sanger Institute (including details of any publication moratoria), please see http://www.sanger.ac.uk/datasharing/"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//' )
    END_VERSIONS
    """
}
