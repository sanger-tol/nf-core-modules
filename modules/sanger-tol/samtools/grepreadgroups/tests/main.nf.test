nextflow_process {

    name "Test Process SAMTOOLS_GREPREADGROUPS"
    script "../main.nf"
    process "SAMTOOLS_GREPREADGROUPS"

    tag "modules"
    tag "modules_sangertol"
    tag "samtools"
    tag "samtools/grepreadgroups"

    test("meles meles - cram") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file(params.modules_testdata_base_path + 'Meles_meles/genomic_data/mMelMel3/hic-arima2/35528_2%231_subset.cram', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assert process.success
            assertAll(
                { assert snapshot(
                    process.out
                ).match() }
            )
        }

    }

    test("meles meles - cram - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file(params.modules_testdata_base_path + 'Meles_meles/genomic_data/mMelMel3/hic-arima2/35528_2%231_subset.cram', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assert process.success
            assertAll(
                { assert snapshot(
                    process.out
                ).match() }
            )
        }

    }

}
