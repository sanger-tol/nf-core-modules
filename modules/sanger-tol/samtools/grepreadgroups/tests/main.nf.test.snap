{
    "meles meles - cram - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "@RG\tID:49240_6#4\tDT:2024-07-25T16:52:42Z\tPU:20240725_LH00275_0063_A22CHVNLT4_6#4\tLB:72611119\tPG:SCS\tSM:SAMEA111361848\tCN:SC\tPL:ILLUMINA\tDS:ERP129860: Sequencing and assembly of symbiont/host genomes in marine and freshwater species, where at least one partner is a microbe, for the Aquatic Symbiosis Genomics project. Upon completion, genomes will be publicly available on an online data coordination center run by the European Bioinformatics Institute. All the data the project generates will be released openly and fairly, as the effort works towards a global community of scientists and ecologists aiming to understand and conserve ocean and freshwater biodiversity. This data is part of a pre-publication release. For information on the proper use of pre-publication data shared by the Wellcome Trust Sanger Institute (including details of any publication moratoria), please see http://www.sanger.ac.uk/datasharing/"
                    ]
                ],
                "1": [
                    "versions.yml:md5,1ec32c4e4eb5a94600c2a439859c11fa"
                ],
                "read_group": [
                    [
                        {
                            "id": "test"
                        },
                        "@RG\tID:49240_6#4\tDT:2024-07-25T16:52:42Z\tPU:20240725_LH00275_0063_A22CHVNLT4_6#4\tLB:72611119\tPG:SCS\tSM:SAMEA111361848\tCN:SC\tPL:ILLUMINA\tDS:ERP129860: Sequencing and assembly of symbiont/host genomes in marine and freshwater species, where at least one partner is a microbe, for the Aquatic Symbiosis Genomics project. Upon completion, genomes will be publicly available on an online data coordination center run by the European Bioinformatics Institute. All the data the project generates will be released openly and fairly, as the effort works towards a global community of scientists and ecologists aiming to understand and conserve ocean and freshwater biodiversity. This data is part of a pre-publication release. For information on the proper use of pre-publication data shared by the Wellcome Trust Sanger Institute (including details of any publication moratoria), please see http://www.sanger.ac.uk/datasharing/"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1ec32c4e4eb5a94600c2a439859c11fa"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-10-24T16:27:27.904507"
    },
    "meles meles - cram": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "@RG\tID:35528_2#1\tDT:2020-11-17T00:00:00+0000\tPU:201117_A00948_0206_AHGML2DSXY_2#1\tLB:33637906\tPG:SCS\tSM:SAMEA5962964\tCN:SC\tPL:ILLUMINA\tDS:ERP116890: Sequencing and assembly of genomes from British species for the Darwin Tree of Life Project. This data is part of a pre-publication release. For information on the proper use of pre-publication data shared by the Wellcome Trust Sanger Institute (including details of any publication moratoria), please see http://www.sanger.ac.uk/datasharing/ "
                    ]
                ],
                "1": [
                    "versions.yml:md5,1ec32c4e4eb5a94600c2a439859c11fa"
                ],
                "read_group": [
                    [
                        {
                            "id": "test"
                        },
                        "@RG\tID:35528_2#1\tDT:2020-11-17T00:00:00+0000\tPU:201117_A00948_0206_AHGML2DSXY_2#1\tLB:33637906\tPG:SCS\tSM:SAMEA5962964\tCN:SC\tPL:ILLUMINA\tDS:ERP116890: Sequencing and assembly of genomes from British species for the Darwin Tree of Life Project. This data is part of a pre-publication release. For information on the proper use of pre-publication data shared by the Wellcome Trust Sanger Institute (including details of any publication moratoria), please see http://www.sanger.ac.uk/datasharing/ "
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1ec32c4e4eb5a94600c2a439859c11fa"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-10-24T16:27:23.996105"
    }
}